src/ldc/libfuzzer.di

```diff
@@ -0,0 +1,63 @@
+/**
+ * Contains convenience functionality for use with libFuzzer.
+ *
+ * Copyright: Authors 2018
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:   LDC Team
+ */
+
+module ldc.libfuzzer;
+
+/**
+ * Defines the necessary code to initialize the D runtime and calls the
+ * FuzzTarget function.
+ * FuzzTarget needs to be a function with signature:
+ *     int function(in ubyte[] data)
+ * The returned value should be 0. Non-zero values are reserved for future
+ * use. See https://llvm.org/docs/LibFuzzer.html#fuzz-target .
+ *
+ * Example:
+ * ---
+ * import ldc.libfuzzer;
+ * mixin LibFuzzerEntryPoint!fuzzMain;
+ *
+ * int fuzzMain(in ubyte[] data)
+ * {
+ *     // Write your test here...
+ *     return 0;
+ * }
+ * ---
+ */
+
+mixin template LibFuzzerEntryPoint(alias FuzzTarget)
+    if (is(typeof(&FuzzTarget) == int function(in ubyte[])))
+{
+    pragma(mangle, "_d_print_throwable") // Work around https://issues.dlang.org/show_bug.cgi?id=12575
+    extern (C) void _d_print_throwable(Throwable t);
+
+    // libFuzzer's user entry point
+    pragma(mangle, "LLVMFuzzerTestOneInput") // Work around https://issues.dlang.org/show_bug.cgi?id=12575
+    extern (C) int LLVMFuzzerTestOneInput(const(ubyte*) data, size_t size)
+    {
+        // D runtime must be initialized, but only once.
+        __gshared static bool runtimeInitialized = false;
+        if (!runtimeInitialized)
```

Comment:
Is this function invoked concurrently in multiple threads? Just wondering if this might be the reason for the spurious fuzzer lit-test failures.

Reply:
No, libFuzzer is single threaded.

Reply:
The `__gshared` results in a small perf boost.